packages/agents/spade_agents/sm_agent.py

```diff
@@ -0,0 +1,89 @@
+from asyncio.futures import Future
+from asyncio.queues import Queue
+from spade.agent import Agent
+from spade.behaviour import CyclicBehaviour, PeriodicBehaviour
+from dataclasses import asdict, dataclass
+import random
+import os
+import json
+import time
+import socket
+
+from spade.message import Message
+
+@dataclass
+class UserState:
+    hp: float
+    mana: float
+    at: float
+
+class SmartWatchAgent(Agent):
+    class StateCollector(PeriodicBehaviour):
+        async def run(self) -> None:
+            state = UserState(hp=random.random(), mana=random.random(), at=random.random())
+            print(f"[[StateCollector]]: Fetching current user state: {state}")
+
+            if self.last_state != state:
+                await self.agent.state_broadcaster.myuser_state.put(state)
+                self.last_state = state
+
+        async def on_start(self) -> None:
+            self.last_state = UserState(0, 0, 0)
+
+    class StateBroadcaster(CyclicBehaviour):
+        def __init__(self):
+            super().__init__()
+            self.myuser_state = Queue()
+
+        async def run(self) -> None:
+            state = await self.myuser_state.get()
+            print(f"[[StateBroadcaster]]: Got state from current user: {state}")
+
+            msg = Message(to=f"broadcast@{os.environ['XMPP']}", metadata={'agent_id': socket.gethostname()})
+            msg.set_metadata('performative', 'inform')
+            msg.body = json.dumps(asdict(state))
+            await self.send(msg)
+            await self.agent.danger_notifier.state_queue.put(state)
+
+    class StateReceiver(CyclicBehaviour):
+        def __init__(self):
+            super().__init__()
+            self.last_time = 0
+            self.user_states = []
+
+        async def run(self) -> None:
+            msg = await self.receive(timeout=100)
+            while not msg:
+                msg = await self.receive(timeout=100)
+
+            state = UserState(**json.loads(msg.body))
+            if msg.metadata['agent_id'] != socket.gethostname():
```

Comment:
@L0czek `socket.gethostname()` returns a container ID and because `from` field is always replaced with `broadcast@xmpp_server` by `mod_broadcast`, we pass a real sender ID in `agent_id` filed in metadata. Also every agent uses the same account on XMPP server.